### PR TITLE
Ranking: add monitoring dashboards for ranking quality

### PR DESCRIPTION
**Pull request by jtibshirani**
This PR updates the monitoring dashboards related to search ranking:
* Replaces the "mean click position" dashboard with two that are more robust. The mean position can be noisy and swayed by outliers. Now, we report the percentage of clicks on the top result, and the percentage within the top 3 (roughly the first page). These represent ranking successes. We exclude searches with very few results, since a click on a top result is not meaningful when there are no other results.
* Corrects an existing dashboard about the distribution of search clicks by result type.
* Adds a new dashboard with the percentage of Zoekt searches that hit the flush time limit. In our experiments on dot com, we found this really helpful in understanding how many searches could be missing relevant results or non-deterministic.

## Test plan

Ran `sg start monitoring` locally, ran example searches, and inspected dashboards.

### CODE REVIEW

**Comment by jtibshirani** [comment]
Here's how it looks with a tiny bit of local data:

Screenshot 2023-03-13 at 5 16 36 PM

(The were no Zoekt searches that hit the time limit, so that's why that panel shows "No data")

File: monitoring/definitions/frontend.go [hunk 1]
```diff
@@ -637,55 +637,60 @@ func Frontend() *monitoring.Dashboard {
...
+							Panel:          monitoring.Panel().LegendFormat("ranked={{ranked}}").Unit(monitoring.Percentage),
+							Owner:          monitoring.ObservableOwnerSearchCore,
+							Interpretation: "The percent of file clicks that were on the first 3 search results, excluding searches with very few results (3 or fewer).",
 						},
 						{
 							Name:        "distribution_of_clicked_search_result_type_over_6h_in_percent",
-							Description: "distribution of clicked search result type over 6h in %",
-							Query:       "round(sum(increase(src_search_ranking_result_clicked_sum{type=\"commit\"}[6h])) / sum (increase(src_search_ranking_result_clicked_sum[6h]))*100)",
+							Description: "distribution of clicked search result type over 6h",
+							Query:       "round(sum(increase(src_search_ranking_result_clicked_count{type=\"repo\"}[6h])) / sum (increase(src_search_ranking_result_clicked_count[6h]))) * 100",
```
Comment (jtibshirani) [inline monitoring/definitions/frontend.go:665]: We were using `sum` here, but I think we want `count`. Am I understanding the intent of this dashboard right?

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (0.00 kb) | 0.00% (+0.01 kb) | 0.00% (+0.01 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/jtibs/ranking-dashboard/compare-report.html;tab=live_object) for a full comparison between the commits 0a083a4 and ef9703e or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: monitoring/definitions/frontend.go (hunk 1 above)
Comment (stefanhengl) [reply monitoring/definitions/frontend.go:665]: I agree, `sum` doesn't make sense.

**Review by stefanhengl — APPROVED** [review]
I wasn't able to start Grafana locally, because I ran into the [same issue you reported](https://github.com/sourcegraph/sourcegraph/issues/48935). How did you run it locally? Approving to unblock and based on the screenshots.

**Comment by jtibshirani** [comment]
> Why do we have 2 "filePathMatch" here?

I tried to debug this but unfortunately couldn't fix it. It's a preexisting issue with that dashboard, which also shows up on dot com. I'll look into it in a follow-up!

> I wasn't able to start Grafana locally, because I ran into the https://github.com/sourcegraph/sourcegraph/issues/48935. How did you run it locally?

I got myself unblocked through a combination of `git clean` and directly running the command to build Grafana images. Let me know if I can help! There's a thread in #ask-dev-experience that we can revive.

**Comment by jtibshirani** [comment]
I added one more dashboard that tracks the total number of search clicks, and rearranged the panels:

Screenshot 2023-03-14 at 10 46 07 AM

Let me know if you have more feedback, I'm happy to make changes in a follow-up.

**Pull request merged**